integrations/aws/main.py

```diff
@@ -0,0 +1,196 @@
+from typing import Any
+
+from port_ocean.core.models import Entity
+from utils import (
+    ACCOUNT_ID_PROPERTY,
+    IDENTIFIER_PROPERTY,
+    KIND_PROPERTY,
+    REGION_PROPERTY,
+    ResourceKindsWithSpecialHandling,
+    _session_manager,
+    describe_accessible_accounts,
+    batch_resources,
+    describe_single_resource,
+    fix_unserializable_date_properties,
+    get_sessions,
+    get_resource_kinds_from_config,
+    resync_cloudcontrol,
+    update_available_access_credentials,
+    validate_request,
+    get_matching_kinds_from_config,
+)
+from port_ocean.context.ocean import ocean
+from loguru import logger
+from starlette.requests import Request
+from port_ocean.core.ocean_types import ASYNC_GENERATOR_RESYNC_TYPE
+
+
+@ocean.on_resync()
+async def resync_all(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
+    await update_available_access_credentials()
```

Comment:
My initial approach was to periodically refresh the credentials, the resync all method that runs before any other resync method seems like a nice compromise, it seems redundant to me to update on evry type of sync, WDYT?

Reply:
for now do it in every reync
nothing promise you it will stay that way
you can update the credentials once and then set it to the event parameters that you have already done this to prevent future calls